### PR TITLE
[wasm][build] Handle `FOUNDATION_BUILD_TOOLS` value set to `OFF`

### PR DESCRIPTION
**Pull request by kateinoigakukun**
This is a follow up to 5e7281b993ca04486ed8f2add39c8b11f132a550, which just adds the `BUILD_TOOLS` option but doesn't actually use it.

### CODE REVIEW

**Review by MaxDesiatov — APPROVED** [review]

**Comment by kateinoigakukun** [comment]
@swift-ci test

File: Sources/CMakeLists.txt [hunk 1]
```diff
@@ -2,4 +2,6 @@ add_subdirectory(UUID)
 add_subdirectory(Foundation)
 add_subdirectory(FoundationNetworking)
 add_subdirectory(FoundationXML)
-add_subdirectory(Tools)
+if(BUILD_TOOLS)
```
Comment (compnerd) [inline Sources/CMakeLists.txt:5]: This is not a standard CMake flag. Can you create a proper `option` for this for something like `FOUNDATION_BUILD_TOOLS`? We can control the default for it to disable it by default on WASM targets if desired.

**Review by compnerd — CHANGES_REQUESTED** [review]

File: Sources/CMakeLists.txt (hunk 1 above)
Comment (kateinoigakukun) [reply Sources/CMakeLists.txt:5]: The option is already declared here as `option` https://github.com/apple/swift-corelibs-foundation/blob/7f62c8c52aeed9a86b503fe7b662c48eda5d0f8a/CMakeLists.txt#L43 but I agree the option name should be prefixed by `FOUNDATION`

**Comment by kateinoigakukun** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Review by compnerd — APPROVED** [review]

**Pull request merged**